webapp/tests/forms/flows/test_step_chooser.py

```diff
@@ -250,6 +251,52 @@ def test_if_no_step_is_valid_next_step_then_return_none(self, step_chooser_witho
         assert returned_next_step is None
 
 
+class TestStepChooserValidateAndUpdateData:
+
+    @pytest.fixture()
+    @pytest.mark.usefixtures('test_request_context')
+    def testing_step_chooser(self):
+        testing_steps = [MockStartStep, MockRenderStep, MockFormWithInputStep, MockYesNoStep, MockFinalStep]
+        step_chooser = StepChooser(title="Testing StepChooser", steps=testing_steps,
+                                        endpoint="lotse", overview_step=MockFormWithInputStep)
+
+        yield step_chooser
+
+    def test_if_not_update_then_return_none_flag_and_unchanged_data(self, app, testing_step_chooser):
+        step_name = MockFormWithInputStep.name
+        update_data = False
+        stored_data = {'name': 'Nagini'}
+        form_data = ImmutableMultiDict({'pet': 'Maledictus', 'date': ['2', '5', '1998'], 'decimal': '100.000'})
+
+        is_valid_flag, updated_data = testing_step_chooser.validate_and_update_data(step_name, update_data, stored_data, form_data)
+
+        assert is_valid_flag is None
+        assert updated_data == stored_data
```

Comment:
I think we should pass a copy of stored_data to the function. Otherwise, this - I think - will never fail?

Same for the two tests below.